include/nbl/builtin/hlsl/workgroup2/arithmetic.hlsl

```diff
@@ -22,11 +22,13 @@ namespace workgroup2
 template<class Config, class BinOp, class device_capabilities=void>
 struct reduction
 {
-    template<class DataAccessor, class ScratchAccessor NBL_FUNC_REQUIRES(ArithmeticDataAccessor<DataAccessor> && ArithmeticSharedMemoryAccessor<ScratchAccessor>)
-    static void __call(NBL_REF_ARG(DataAccessor) dataAccessor, NBL_REF_ARG(ScratchAccessor) scratchAccessor)
+    using scalar_t = typename BinOp::type_t;
+
+    template<class ReadOnlyDataAccessor, class ScratchAccessor NBL_FUNC_REQUIRES(ArithmeticDataAccessor<ReadOnlyDataAccessor> && ArithmeticSharedMemoryAccessor<ScratchAccessor>)
```

Comment:
you need a ReadDataAccessor concept, so it doesn't check the `set` method existence